library/core/src/ptr/non_null.rs

```diff
@@ -1803,4 +1809,66 @@ mod verify {
         let maybe_null_ptr =  if kani::any() { xptr as *mut i32 } else { null_mut() };
         let _ = NonNull::new(maybe_null_ptr);
     }
+
+    #[kani::proof_for_contract(NonNull::len)]
+    pub fn non_null_check_len() {
+        const SIZE: usize = 20000;
+        // Create a non-deterministic array of size SIZE
+        let arr: [i8; SIZE] = kani::any();  
+        // Get a raw pointer to the array
+        let raw_ptr: *const i8 = arr.as_ptr();  
+        // Create a NonNull slice from the raw pointer
+        let non_null_slice: NonNull<[i8]> = unsafe { NonNull::slice_from_raw_parts(NonNull::new(raw_ptr as *mut i8).unwrap(), SIZE) };  
+
+        // Perform the length check
+        let len = non_null_slice.len();
+        assert!(len == SIZE);
+    }
+
+    #[kani::proof_for_contract(NonNull::is_empty)]
+    pub fn non_null_check_is_empty() {
+        const SIZE: usize = 0;
+        // Create a non-deterministic array of size SIZE
+        let arr: [i8; SIZE] = kani::any();  
+        // Get a raw pointer to the array
+        let raw_ptr: *const i8 = arr.as_ptr();  
+        // Create a NonNull slice from the raw pointer
+        let non_null_slice: NonNull<[i8]> = unsafe { NonNull::slice_from_raw_parts(NonNull::new(raw_ptr as *mut i8).unwrap(), SIZE) };  
+
+        // Perform the is_empty check
+        let is_empty = non_null_slice.is_empty();
+        assert!(is_empty);
+    }
+
+    #[kani::proof_for_contract(NonNull::is_aligned)]
+    pub fn non_null_slice_is_aligned_check() {
+        // Create a non-deterministic integer
+        let mut data: i32 = kani::any();  
+        // Get a raw pointer to the integer
+        let raw_ptr: *mut i32 = &mut data;
+        // NonNull pointer to the integer
+        let ptr = unsafe { NonNull::new(raw_ptr).unwrap() };
+
+        // Perform the alignment check
+        let result = ptr.is_aligned();
+        assert!(result);
```

Comment:
Add negative tests where pointer is not aligned and assert result

Reply:
You don't necessarily need to assert result since you have already added that to the contract. Note that the post-condition of the contract will already be encoded as an assertion.